lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -46,17 +46,22 @@ export class TokenCache implements ITokenCache {
      * @param request
      * @param response
      * @param options
+     * @returns A `CacheRecord` containing the entities that were loaded.
      */
-    loadExternalTokens(request: SilentRequest, response: ExternalTokenResponse, options: LoadTokenOptions): void {
+    loadExternalTokens(request: SilentRequest, response: ExternalTokenResponse, options: LoadTokenOptions): CacheRecord {
         this.logger.info("TokenCache - loadExternalTokens called");
 
         if (!response.id_token) {
             throw BrowserAuthError.createUnableToLoadTokenError("Please ensure server response includes id token.");
         }
 
+        const cacheRecord = new CacheRecord();
+
         if (request.account) {
-            this.loadIdToken(response.id_token, request.account.homeAccountId, request.account.environment, request.account.tenantId, options);
-            this.loadAccessToken(request, response, request.account.homeAccountId, request.account.environment, request.account.tenantId, options);
+            cacheRecord.account = this.loadAccount(response.id_token, request.account.environment, undefined, undefined, request.account.homeAccountId);
```

Comment:
This looks great, if I could offer a final nitpick it'd be to do this in a single statement (or 2 if you need to have the account entity first for the following method calls):

```suggestion
            cacheRecordAccount = this.loadAccount(response.id_token, request.account.environment, undefined, undefined, request.account.homeAccountId);
            
            cacheRecord = {
                        account: cacheRecordAccount,
                        idToken: this.loadIdToken(response.id_token, cacheRecordAccount.homeAccountId, request.account.environment, request.account.tenantId),
                        accessToken: this.loadAccessToken(request, response, cacheRecordAccount.homeAccountId, request.account.environment, request.account.tenantId, options);
                       refreshToken: this.loadRefreshToken(request, response, cacheRecordAccount.homeAccountId, request.account.environment)
                       }
```

Or something similar.

Reply:
updated